src/ui/components/Library/Team/View/TestRuns/suspense/TestRunsCache.ts

```diff
@@ -61,12 +61,29 @@ export const testRunDetailsCache = createCache<
     const testsWithRecordings =
       testRunNode?.tests.map<TestRunTestWithRecordings>(test => {
         durationMs += test.durationMs;
-        const recs = orderBy(test.recordings.map(convertRecording), "date", "desc");
-        recordings.push(...recs);
 
         return {
           ...test,
-          recordings: recs,
+          executions: test.executions.map(e => {
+            const recs = e.recordings.map(convertRecording);
+            recordings.push(...recs);
+
+            // Adapt the execution status to convert "failed" execution status
+            // to "flaky" when the test eventually passes
+            let result = e.result;
+            if (test.result === "flaky") {
+              if (testFailed(e)) {
+                result = "flaky";
+              } else if (testPassed(e)) {
+                result = "passed";
+              }
+            }
```

Comment:
Moving the failed vs flaky logic out of the client feels kind of like taking a step forward in one place and a step backward in another place, if we have to deconstruct it like this.

Reply:
I was taking a short cut to show the expected UI for flaky tests by adjusting the status so the UI didn't have to know if the test was "flaky" to show the flaky icon for a failed test. If you'd prefer, I can move a version of this `if` to the view component instead.

Reply:
I don't know that I feel strongly about where it lives. Just remarking on the fact that the frontend _used_ to have logic to do this.